fingerprintdb_cuda.h

```diff
@@ -76,20 +107,24 @@ class FingerprintDB
     char* getSmiles(int index) const { return m_smiles[index]; }
     char* getID(int index) const { return m_ids[index]; }
 
-    size_t getFingerprintDataSize() const { return m_data_size; };
+    size_t getFingerprintDataSize() const { return m_total_data_size; };
     int getFingerprintBitcount() const { return m_fp_intsize * sizeof(int) * 8; }
 
+    void search_storage(const Fingerprint& query,
+            const std::shared_ptr<FingerprintDBStorage>& storage,
+            std::vector<SortableResult>* sortable_results,
+            unsigned int return_count, float similarity_cutoff) const;
+
   protected:
     // INTERNAL:  A CPU implementation of tanimoto similarity for validation
     float tanimoto_similarity_cpu(const Fingerprint& fp1,
                                   const Fingerprint& fp2) const;
-
     std::vector<int> fold_data(const std::vector<int>& unfolded) const;
 
-    std::vector<int> m_data, m_folded_data;
-    std::shared_ptr<FingerprintDBPriv> m_priv; // Used to conceal cuda types
-    int m_count, m_fp_intsize, m_fold_factor;
-    size_t m_data_size;
+    std::vector<std::shared_ptr<FingerprintDBStorage> > m_storage;
+
+    int m_total_count, m_fp_intsize, m_fold_factor;
```

Comment:
Can these be on separate lines? I guess I really want you to clang-format the whole thing.

Reply:
I can run that as a separate later commit...